core/cap-0046-07.md

```diff
@@ -419,30 +545,30 @@ As a consequence the final formula looks like this:
 
-##### Putting it together
+We can simplify this even further by replacing the exponential component by a steep linear slope that causes fees to be "extremely high" at `M_buffer`, which turns the formula into what is specified above:
 
-"read/write" operations need to first read data before writing it. The amount of data written back can be larger or smaller than what was read, as consequence:
-* The number of ledger entry reads is the size of ledger entries referenced in ledger footprints (both read and read/write).
-* The number of bytes to read is the size of bucket entries from both the read and read/write footprints.
-* The number of bytes to write is the number of bytes associated with bucket entries referenced by the readWrite footprint.
-* The number of ledger entry to write is the size of the read/write footprint.
+`fee_rate(s) = (feeRateM - feeRate)*s/M_base + feeRate + if (s > M_base, K*(s-M_base)/B_buffer, 0)`
 
-##### Effective fee and flooding
+where `K >= 1`.
 
-Having the fee model depend on ledger size creates some complication when trying to reason about multiple transactions getting applied: a naive solution would just try to follow the price curve exactly, causing fees to evolve on a per transaction basis.
-This would create incentives to front-run transactions within the same transaction set, and would also make it hard to decide if a transaction should get flooded.
+##### Ledger size averaging
 
-In the context of this proposal, we make the following observation: assuming that validators pick a small upper bound for the total number of bytes that can be added to the bucket list per ledger relative to the bucket list size, we can then assume that the price of storage varies marginally per ledger (ie the price paid by the first transaction is not that different from the last transaction in a ledger), and that the reference price still "resets" to the proper price every ledger (as it's based on the size of the bucket list).
+Tracking the ledger size for every ledger introduces unnecessary noise that leads to the following issues:
+* flooding might be somewhat imprecise due to fees changing every ledger with a risk of transaction becoming invalid
+* wrong incentives, such as trying to pay the rent for a long time period right after the bucket list merge ledger
+* fee estimations are harder for the clients
 
-As a consequence, we can just price transactions independently, based on the bucket list from the last closed ledger, and allocation within a transaction (for multiple bytes) is a flat rate as well.
+To alleviate all of these issues, instead of using the current ledger size, this proposal uses the average of the ledger size over the sliding window, that is large enough to average out most of the noise coming from short-term merges and rather representing the ledger size change trends rather than actual side at any moment.
```

Comment:
```suggestion
To alleviate all of these issues, instead of using the current ledger size, this proposal uses the average of the ledger size over the sliding window, that is large enough to average out most of the noise coming from short-term merges and rather representing the ledger size change trends rather than actual size at any moment.
```